javascript/javascript3/week3/promise-classmates.js

```diff
@@ -0,0 +1,47 @@
+//Getting into promises
+
+
+function promiseToClassmates(userName){
+    return new Promise((resolve, reject) => {
+        fetch(`https://api.github.com/search/repositories?q=user:${userName}`)
+        .then(res=>res.json()) 
+        .then(data=>{
+            console.log(data);
+            renderGitRepoData(data, userName);
+            console.log("all promised has resolved")
+        resolve(data)})
+        .catch((e)=> reject(e))
```

Comment:
The point of Promise.all is that you can handle resolve and reject for every instance together. In addition to that, [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is an asynchronous function that returns a promise by default, there's no need to create a new promise. 

So what you can do is to have this promise chain directly after the Promise.all, something like this:

```
    return Promise.all([fetchClassmates("simonreddy2001"),
        fetchClassmates("semih1239"),
        fetchClassmates("Jul-S")
    ]).then(responseArray => Promise.all(responseArray.map(response => response.json()))).then(dataArray => {dataArray.forEach(data => renderGitRepoData(data.items))}).catch((e)=> reject(e));
```

Reply:
I fixed it and thanks for good explanation.